### ISSUE TITLE
 Branches are not counted

### ISSUE THREAD

**Issue by Corje** [Kotlin/kotlinx-kover#4]
Reproduced with: 'org.jetbrains.kover-plugin' version '0.4', intellij-coverage-agent-1.0.596.jar, Kotlin JVM Gradle plugin 1.6.0-RC-303, gradle-7.1

```kotlin
enum class BranchBug {
    ONE, TWO, THREE
}

fun checkWhenBranches(b: BranchBug) = when (b) {
    BranchBug.ONE -> "one"
    BranchBug.TWO -> "two"
    BranchBug.THREE -> "three"
}
```
Run `test` task. Part from a generated report, branches are 0:
```xml
<class name="one.branchbug.BranchBugKt" sourcefilename="BranchBug.kt">
<method name="checkWhenBranches" desc="(Lone/branchbug/BranchBug;)Ljava/lang/String;">
<counter type="LINE" covered="2" missed="2"/>
<counter type="BRANCH" covered="0" missed="0"/>
</method>
</class>
```
Branches are counted in JaCoCo report for the same case:
```xml
<class name="one/branchbug/BranchBugKt" sourcefilename="BranchBug.kt">
			<method name="checkWhenBranches" desc="(Lone/branchbug/BranchBug;)Ljava/lang/String;" line="7">
				<counter type="INSTRUCTION" missed="4" covered="12"/>
				<counter type="BRANCH" missed="2" covered="1"/>
				<counter type="LINE" missed="1" covered="3"/>
				<counter type="COMPLEXITY" missed="2" covered="1"/>
				<counter type="METHOD" missed="0" covered="1"/>
			</method>
			<counter type="INSTRUCTION" missed="4" covered="12"/>
			<counter type="BRANCH" missed="2" covered="1"/>
			<counter type="LINE" missed="1" covered="3"/>
			<counter type="COMPLEXITY" missed="2" covered="1"/>
			<counter type="METHOD" missed="0" covered="1"/>
			<counter type="CLASS" missed="0" covered="1"/>
		</class>
```
Also branch counter is always present in a report even when there are no branches in method (unlike JaCoCo).

**Comment by Corje** [comment]
In `kotlinx-kover` version `0.2.2` branches are counted (like in tracing mode of Intellij agent).
However branch counter is still presented even when there are no branches (0 missed, 0 covered), probably this can be omitted.

**Comment by shanshin** [comment]
The IntelliJ agent counts branches incorrectly.

For class
```
class MyBranchedClass {
    fun foo(value: Int) {
        if (value < 0) {
            println("LE")
        } else if (value == 0) {
            println("EQ")
        } else {
            println("GE")
        }
    }
}
```
and test
```
    @Test
    fun testBranches() {
        MyBranchedClass().foo(-20)
    }
```

Intellig Agent generates XML report
```
<method name="foo" desc="(I)V">
<counter type="INSTRUCTION" missed="3" covered="2"/>
<counter type="BRANCH" missed="1" covered="0"/>
<counter type="LINE" missed="3" covered="2"/>
</method>
```

at the same time JaCoCo generates
```
            <method name="foo" desc="(I)V" line="5">
                <counter type="INSTRUCTION" missed="17" covered="11"/>
                <counter type="BRANCH" missed="3" covered="1"/>
                <counter type="LINE" missed="3" covered="3"/>
                <counter type="COMPLEXITY" missed="2" covered="1"/>
                <counter type="METHOD" missed="0" covered="1"/>
            </method>
```
which is more correct

**Comment by shanshin** [comment]
Fixed in `0.5.0-RC2`
